central/notifiers/generic/generic.go

```diff
@@ -250,6 +246,11 @@ func (g *generic) AuditLoggingEnabled() bool {
 }
 
 func (g *generic) getPassword() (string, error) {
+	if g.GetGeneric().GetUsername() == "" {
```

Comment:
Not sure why this check is here?

Reply:
That check is to short circuit and return `""` as password. In generic notifier, empty username and password mean that it is unauthenticated. If it is not short circuited here, it will go into the codec functions, which would error out if they receive an empty string.